src/test/utils.test.ts

```diff
@@ -0,0 +1,93 @@
+import {buildAllPairs} from "../utils";
+
+test('buildAllPairs creates correlation pairs without missing any fields and without duplicates', () => {
+    const input = ["apple", "banana", "strawberries", "pears", "grapefruit", "bread", "milk"]
+    const results = buildAllPairs(input)
+    expect(results.length).toEqual(21);
```

Comment:
Maybe make this `expect(results.length).toEqual(input.length ** input.length)` in case we change the input

Reply:
Yes, I can change this. It would be input.length choose 2.

Reply:
Right, right. Maths